libs/application/templates/new-primary-school/src/forms/NewPrimarySchoolForm/primarySchoolSection/newSchoolSubSection.ts

```diff
@@ -61,11 +70,11 @@ export const newSchoolSubSection = buildSubSection({
                       gradeLevels?.includes(childGradeLevel),
                     )?.length > 0,
                 )
-                ?.map(({ name }) => ({
-                  value: name,
+                ?.map(({ name, unitId }) => ({
+                  value: unitId || name || '',
```

Comment:
Do I need to check the name too? 
```suggestion
                  value: unitId || '',
```

Reply:
Fixed